internal/pkg/template/workload.go

```diff
@@ -522,11 +511,18 @@ func (cfg *ALBListener) RulePaths() []string {
 	return rulePaths
 }
 
-// ServiceConnect holds configuration for ECS Service Connect.
-type ServiceConnect struct {
+// ServiceConnectServer defines the container name and port which a service routes Service Connect through.
+type ServiceConnectServer struct {
+	Name  string
+	Port  string
 	Alias *string
 }
 
+// IsExposed returns true if the ServiceConnectServer has an accessible port to receive traffic.
+func (opts ServiceConnectServer) IsExposed() bool {
+	return opts.Port != "" && opts.Port != NoExposedContainerPort
```

Comment:
oh interesting! `ServiceConnectServer != nil` doesn't mean that the server is truly enabled. For a service to _really_ be a sc server, its `ServiceConnectServer` needs to be non-nil, AND its port needs to be `NoExposedContainerPort`.

I wonder if we can move this logic to transformer - that if the target port is `NoExposedContainerPort`, then we assign `nil` to `ServiceConnectServer`.